src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -358,23 +370,139 @@ def setClean(self, wflow):
         wflow['IsClean'] = self._checkClean(wflow)
         return wflow
 
+    def _checkLogDBClean(self, wflow):
+        """
+        An auxiliary function used to only check the LogDB cleanup status.
+        It makes a query to LogDB in order to verify there are no any records for
+        the current workflow
+        :param wflow:       A MSRuleCleaner workflow representation
+        :return:            Bool
+        """
+        cleanStatus = False
+        logDBRecords = self.logDB.get(wflow['RequestName'])
+        self.logger.debug("logDBRecords: %s", pformat(logDBRecords))
+        if logDBRecords:
+            cleanStatus = True
+        return cleanStatus
+
+    def setLogDBClean(self, wflow):
+        """
+        A function to set the 'IsLogDBClean' flag based on the presence of any
+        records in LogDB for the current workflow.
+        :param  wflow:      A MSRuleCleaner workflow representation
+        :return wflow:
+        """
+        wflow['IsLogDBClean'] = self._checkLogDBClean(wflow)
+        return wflow
+
+    def findTargStatus(self, wflow):
+        """
+        Find the proper targeted archival status
+        :param  wflow:      A MSRuleCleaner workflow representation
+        :return wflow:
+        """
+        # Check the available status transitions before we decide the final status
+        targetStatusList = RequestStatus.REQUEST_STATE_TRANSITION.get(wflow['RequestStatus'], [])
+        targetStatusReg = re.compile(r'.*archived')
```

Comment:
How about making it an object variable and compile it only once?

Reply:
I was also thinking the same and hesitating initially. But now I am sure. 
Done now.